src/events/SDL_keyboard.c

```diff
@@ -585,20 +585,40 @@ SDL_SetKeymap(int start, SDL_Keycode * keys, int length)
 {
     SDL_Keyboard *keyboard = &SDL_keyboard;
     SDL_Scancode scancode;
+    SDL_bool num_keycode_exists[10] = {SDL_FALSE};
+    SDL_bool all_numbers_present = SDL_TRUE;
+    int i;
 
     if (start < 0 || start + length > SDL_NUM_SCANCODES) {
         return;
     }
 
     SDL_memcpy(&keyboard->keymap[start], keys, sizeof(*keys) * length);
 
-    /* The number key scancodes always map to the number key keycodes.
+    /* If any number key keycode is missing,
+     * the number key scancodes map to the number key keycodes.
      * On AZERTY layouts these technically are symbols, but users (and games)
      * always think of them and view them in UI as number keys.
+     * But on one-handed dvorak layouts most number keys are on lower rows,
+     * while the top row has some ASCII letters.
```

Comment:
Hmm, I wonder whether changing the given keymap is the right thing to do at all.

Most of the major backends (X11, Wayland, Cocoa, Win32) create their keymap by calling OS APIs to get the (theoretically) correct mapping. Presumably some/all of those backends handle AZERTY and DVORAK properly and don't want part of their mapping clobbered by the keyboard core thinking it knows better.

If I understand the change correctly, it seems like that's exactly what's occurring here and breaking these DVORAK layouts. The existing mapping from the platform backend is correct, then we overwrite the mapping for what we think are the number keys.

Reply:
Seems it was "fix" for French/Belgian layouts on non-Windows platfroms to behave like Windows with these layouts: https://github.com/libsdl-org/SDL/commit/3b837a26591fe9146353d607ce497fd7a5641290
But it broke Dvorak layouts.

Also have found same "fix" for 0-9 keys [in Windows keymap code](https://github.com/libsdl-org/SDL/blob/main/src/video/windows/SDL_windowskeyboard.c#L130-L136) but I cannot find rationale and original changelist number.

Reply:
Thanks for pointing out the handling in SDL_windowskeymap.c @DJm00n, I didn't expect there to be any in OS-specific code. and 
The Windows keymap code predates the generic handling in SDL_keyboard.c (which was added to make the behavior consistent), so I assume it's no longer necessary.
But I don't have a Windows machine to test it on.